### PR TITLE
Enable MinGW builds 

### PR DESCRIPTION
**Pull request by Amphaal**
Requires uasm (https://github.com/Terraspace/UASM) to compile Windows ASM files, which is available through MSYS2 depos.

### CODE REVIEW

File: compat/mingw/dbghelp.h [hunk 1]
```diff
@@ -0,0 +1,249 @@
+// Copyright 2014 The Crashpad Authors. All rights reserved.
```
Comment (Swatinem) [inline compat/mingw/dbghelp.h]: Are these files upstream? I wonder why they are not part of our fork?

File: compat/mingw/dbghelp.h [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CRASHPAD_COMPAT_MINGW_DBGHELP_H_
+#define CRASHPAD_COMPAT_MINGW_DBGHELP_H_
+
+#include_next <dbghelp.h>
```
Comment (Swatinem) [inline compat/mingw/dbghelp.h]: TIL `include_next` exists :-)

File: compat/mingw/dbghelp.h (hunk 1 above)
Comment (Amphaal) [reply compat/mingw/dbghelp.h]: Ah, this is a patch to the dbghelp.h header file of MinGW, which misses somes references required by crashpad. I just copy-pasted the whole file from /compat/non_win/dbghelp.h and removed the duplicated references :P

File: compat/mingw/dbghelp.h (hunk 2 above)
Comment (Amphaal) [reply compat/mingw/dbghelp.h]: ...and you taught me "TIL" ! ;)

**Comment by jan-auer** [comment]
@Swatinem please squash-merge this when you're fine with it.

Just as a word of caution, since this adds significationly more modifications, this will make it harder to track Crashpad. Without looking into the details - should these changes potentially be fixed in upstream crashpad (excluding the CMake changes, of course).

**Comment by Amphaal** [comment]
> @Swatinem please squash-merge this when you're fine with it.
> 
> Just as a word of caution, since this adds significationly more modifications, this will make it harder to track Crashpad. Without looking into the details - should these changes potentially be fixed in upstream crashpad (excluding the CMake changes, of course).

You are right, but they might want their building process updated accordingly for testing purposes. We should definitively poke the Crashpad team at some point so they can check our work on this.

**Comment by Amphaal** [comment]
> @Swatinem please squash-merge this when you're fine with it.
> 
> Just as a word of caution, since this adds significationly more modifications, this will make it harder to track Crashpad. Without looking into the details - should these changes potentially be fixed in upstream crashpad (excluding the CMake changes, of course).

You are right, but they might want their building process updated accordingly for testing purposes. We should definitely poke the Crashpad team at some point so they can check our work on this.

**Comment by Swatinem** [comment]
> We should definitely poke the Crashpad team at some point so they can check our work on this.

Please do so. I merged this, to unblock further work on `sentry-native`, but would be nice to actually move this upstream as well. Which reminds me that we should maybe also mare an upstream merge at some point.

**Comment by Amphaal** [comment]
https://groups.google.com/a/chromium.org/forum/#!topic/crashpad-dev/Y2tt6dDNIBI here we go. 

> Which reminds me that we should maybe also mare an upstream merge at some point.

That's a great idea !

**Pull request merged**